src/platforms/evdev-rs/platform_bridge.h

```diff
@@ -0,0 +1,142 @@
+/*
+ * Copyright © Canonical Ltd.
+ *
+ * This program is free software: you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License version 2 or 3,
+ * as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef MIR_INPUT_EVDEV_RS_PLATFORM_BRIDGE_H
+#define MIR_INPUT_EVDEV_RS_PLATFORM_BRIDGE_H
+
+#include <mir/console_services.h>
+#include <mir/input/input_device.h>
+#include <mir/input/event_builder.h>
+#include <memory>
+
+namespace mir
+{
+class Device;
+
+namespace input
+{
+namespace evdev_rs
+{
+class Platform;
+
+/// Wraps a #mir::Device so that it can be kept alive in rust.
+///
+/// The platform must maintain a reference to the #mir::Device
+/// or else the device will be destroyed.
+class DeviceBridgeC
+{
+public:
+    DeviceBridgeC(std::unique_ptr<Device> device, int fd);
+    int raw_fd() const;
+
+private:
+    std::unique_ptr<Device> device;
+    int fd;
+};
+
+class EventBuilderWrapper
+{
+public:
+    explicit EventBuilderWrapper(EventBuilder* event_builder);
+    std::shared_ptr<MirEvent> pointer_event(
+        bool has_time,
+        uint64_t time_microseconds,
+        int32_t action,
+        uint32_t buttons,
+        bool has_position,
+        float position_x,
+        float position_y,
+        float displacement_x,
+        float displacement_y,
+        int32_t axis_source,
+        float precise_x,
+        int discrete_x,
+        int value120_x,
+        bool scroll_stop_x,
+        float precise_y,
+        int discrete_y,
+        int value120_y,
+        bool scroll_stop_y
+    ) const;
+
+    std::shared_ptr<MirEvent> key_event(
+        bool has_time,
+        uint64_t time_microseconds,
+        int32_t action,
+        uint32_t scancode
+    ) const;
+
+private:
+    EventBuilder* event_builder;
+};
+
+class RectangleC
```

Comment:
Is this "C" suffix a new naming convention? Or part of the translation layer mechanics?

Reply:
I haven't yet seen the need for such a distinction, but I've not delved far enough into the code to be confident there isn't one